dataset/src/main/scala/frameless/implicits.scala

```diff
@@ -32,4 +32,12 @@ object implicits {
     // implicit def floatToDouble[T](col: TypedColumn[T, Float]): TypedColumn[T, Double] = col.cast[Double]
     // implicit def floatToBigDecimal[T](col: TypedColumn[T, Float]): TypedColumn[T, BigDecimal] = col.cast[BigDecimal]
   }
+
+  object injections {
+    implicit val javaBoolean: Injection[java.lang.Boolean, Boolean] = Injection(a => a, b => b)
```

Comment:
DId you consider `Injection[java.lang.Boolean, Option[Boolean]]`? Often when I see Java boxed primitives, it's Java-style Scala code using nulls.